applications/ShapeOptimizationApplication/python_scripts/algorithm_bead_optimization.py

```diff
@@ -112,6 +118,15 @@ def InitializeOptimizationLoop(self):
         self.mapper = mapper_factory.CreateMapper(self.design_surface, self.design_surface, self.mapper_settings)
         self.mapper.Initialize()
 
+        if  self.filter_penalty_term:
+            if self.algorithm_settings["penalty_filter_radius"].GetDouble() != self.mapper_settings["filter_radius"].GetDouble() :
```

Comment:
Double comparison with ==    -,-

Reply:
@philbucher What do you mean by this?

Reply:
Here i compare to doules explicitly instead of checking for a deviation below a tolerance.